dev/TabView/TabView.h

```diff
@@ -8,36 +8,70 @@
 
 #include "TabView.g.h"
 #include "TabView.properties.h"
-#include "TabViewTabClosingEventArgs.g.h"
-#include "TabViewTabDraggedOutsideEventArgs.g.h"
+#include "TabViewTabCloseRequestedEventArgs.g.h"
+#include "TabViewTabDroppedOutsideEventArgs.g.h"
+#include "TabViewTabDragStartingEventArgs.g.h"
+#include "TabViewTabDragCompletedEventArgs.g.h"
 #include "DispatcherHelper.h"
 
-class TabViewTabClosingEventArgs :
-    public winrt::implementation::TabViewTabClosingEventArgsT<TabViewTabClosingEventArgs>
+class TabViewTabCloseRequestedEventArgs :
+    public ReferenceTracker<TabViewTabCloseRequestedEventArgs, winrt::implementation::TabViewTabCloseRequestedEventArgsT, winrt::composing, winrt::composable>
 {
 public:
-    TabViewTabClosingEventArgs(winrt::IInspectable const& item) { m_item = item; }
+    TabViewTabCloseRequestedEventArgs(winrt::IInspectable const& item, winrt::TabViewItem tab) { m_item = item; m_tab.set(tab); }
 
-    bool Cancel() { return m_cancel; }
-    void Cancel(bool value) { m_cancel = value; }
+    winrt::IInspectable Item() { return m_item; }
+    winrt::TabViewItem Tab() { return m_tab.get(); }
+
+private:
+    winrt::IInspectable m_item{};
+    tracker_ref<winrt::TabViewItem> m_tab{ this };
+};
+
+class TabViewTabDroppedOutsideEventArgs :
+    public ReferenceTracker<TabViewTabDroppedOutsideEventArgs, winrt::implementation::TabViewTabDroppedOutsideEventArgsT, winrt::composing, winrt::composable>
+{
+public:
+    TabViewTabDroppedOutsideEventArgs(winrt::IInspectable const& item, winrt::TabViewItem tab) { m_item = item; m_tab.set(tab); }
+
+    winrt::IInspectable Item() { return m_item; }
+    winrt::TabViewItem Tab() { return m_tab.get(); }
+
+private:
+    winrt::IInspectable m_item{};
+    tracker_ref<winrt::TabViewItem> m_tab{ this };
+};
 
+class TabViewTabDragStartingEventArgs :
+    public ReferenceTracker<TabViewTabDragStartingEventArgs, winrt::implementation::TabViewTabDragStartingEventArgsT, winrt::composing, winrt::composable>
+{
+public:
+    TabViewTabDragStartingEventArgs(winrt::DragItemsStartingEventArgs const& args, winrt::IInspectable const& item, winrt::TabViewItem tab) { m_args = args; m_item = item; m_tab.set(tab); }
+
```

Comment:
Wrap lines, and [prefer initialization to assignment](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c49-prefer-initialization-to-assignment-in-constructors):

```c++
: m_args(args), m_item(item), m_tab(tab)
```

Reply:
OK, done.